CONTRIBUTING.md

```diff
@@ -29,6 +29,43 @@ integration tests that need a running synapse instance. These tests reside in
 [README](./testing/matrix-sdk-integration-testing/README.md) to easily set up a
 synapse for testing purposes.
 
+## Commit messages and PR title guidelines
+
+Ideally, a PR should have a *proper title*, with *atomic logical commits*, and each commit
+should have a *good commit message*.
+
+An *atomic logical commit* is one that is ideally small, can be compiled in isolation, and pass
+tests. This is useful to make the review process easier (help your reviewer), but also when running
+bisections, helping identifying which commit introduced a regression.
+
+A *good commit message* should be composed of:
+
+- a hint to which area/feature is related by the commit
+- a short description that would give sufficient context for a reviewer to guess what the commit is
+  about.
+
+Examples of commit messages that aren't so useful:
+
+- “add new method“
+- “enhance performance“
+- “fix receipts“
+
+Examples of good commit messages:
+
+- “ffi: add new method `frobnicate_the_foos`”
+- “indexeddb: break up the request inside `get_inbound_group_sessions`”
+- “read_receipts: store receipts locally, fixing #12345”
```

Comment:
```suggestion
- “ffi: Add new method `frobnicate_the_foos`”
- “indexeddb: Break up the request inside `get_inbound_group_sessions`”
- “read_receipts: Store receipts locally, fixing #12345”
```

Not just me who prefers this, right?

Reply:
I approve this proposal.

Reply:
I don't prefer it, both work for me, and I won't blame anyone who's not putting it uppercase, but yeah we can still show it in examples if you have a strong opinion here.